static/docs/user-guide/install/completion.md

```diff
@@ -0,0 +1,154 @@
+# Shell Tab Completion
+
+Enjoy working with DVC faster and with fewer typos!
+
+## How it works
+
+Command completion is usually requested by pressing the `tab` key on your shell,
+it will then present the possible options that could follow that command call.
+For example:
+
+```dvc
+$ dvc r # Press [tab] key
+ -- dvc commands --
+remote  -- Manage remote storage configuration.
+remove  -- Remove outputs of DVC-file.
+repro   -- Check for changes and reproduce DVC-file and dependencies.
+root    -- Relative path to project's directory.
+run     -- Generate a stage file from a command and execute the command.
+```
+
+Depending on what you typed on the command line so far, it completes:
+
+- Available DVC commands.
+- Options that are available for a particular command.
+- File names that make sense in a given context, such as using them as a target
+  for some commands.
+- Arguments for selected options. For example, `dvc repro` completes with stage
+  files to reproduce.
+
+## Which shell you have
+
+Use the command `echo $0` to check your active shell, if you are not sure which
+one you are using (Bash or Zsh).
+
+Depending on the shell that you are using, follow the instruction below to
+install and configure shell completion.
+
+## Bash completion on Mac OS
+
+1. First, make sure that Bash completion support is installed:
+
+   ```dvc
+   $ brew install bash-completion
+   ```
+
+2. Then download the
+   [DVC completion script](https://github.com/iterative/dvc/blob/master/scripts/completion/dvc.bash)
+   and save it on `/usr/local/etc/bash_completion.d/`:
+
+   ```dvc
+   $ sudo wget \
+       -O /usr/local/etc/bash_completion.d/dvc \
+       https://raw.githubusercontent.com/iterative/dvc/master/scripts/completion/dvc.bash
+   ```
+
+3. Edit `~/.bash_profile` and make sure that these lines are there:
+
+   ```bash
+   if [ -f $(brew --prefix)/etc/bash_completion ]; then
+       . $(brew --prefix)/etc/bash_completion
+   fi
+   ```
+
+4. Finally, `source ~/.bash_profile` or launch a new terminal to activate it.
+
+## Bash completion on Ubuntu
+
+On a Linux system bash-completion should work out of the box, but if you are
```

Comment:
is it true, @efiop ? or is it only if you install it as a binary?

Reply:
@shcheklein only if you install from our deb package.

Reply:
Also not sure what "VPS" has to do with anything here. @dashohoxha Could you clarify?

Reply:
> Also not sure what "VPS" has to do with anything here. @dashohoxha Could you clarify?

The Ubuntu server image in some cloud providers is minimized (to save time in deployment), and the package `bash-completion` is missing. More details here: https://linuxhandbook.com/enable-tab-completion/

Reply:
@dashohoxha so, should we update to clarify that it's not out of the box in certain cases? And I still don't understand what VPS has do with it.

Reply:
> should we update to clarify that it's not out of the box in certain cases

I think the explanation is correct because it says "In any case, follow these steps to make sure that it is installed". So, if it works already that's fine, if not follow these steps to make sure that everything is setup correctly.

> I still don't understand what VPS has do with it.

If you have installed DVC in a VPS on the cloud, the package `bash-completion` probably is not installed, and without it tab completion cannot work. This is one of the cases when bash completion does not out of the box.

Maybe we should not mention it at all and just say: "In case that bash completion does not work, follow these steps...". What is your take on it?

Reply:
Explanation is might be correct, but the statement `On a Linux system bash-completion should work out of the box` is not. It's just false and misleading.

I get the cloud part, I don't understand what _VPS_ has to do with it. Could you elaborate? Give an example?

Reply:
> I get the cloud part, I don't understand what VPS has to do with it. Could you elaborate? Give an example?

I removed the paragraph that mentions VPS because it seems superfluous. Anyway, if you are confused by it, I can imagine that a normal user would be much more confused. I hope that now it is simpler and more logical.

Reply:
Yep, thanks. I would say you removed too much. I would keep a mention that it should be working out of the box on Linux if it's installed from the package (manually or from the custom repository).

Reply:
>  I would keep a mention that it should be working out of the box on Linux if it's installed from the package (manually or from the custom repository).

Installing from the package still does not guarantee that bash completion should work out of the box. If you install it on a VPS on the cloud this would not be enough.